services/QuillLMS/app/controllers/api/v1/focus_points_controller.rb

```diff
@@ -34,8 +37,13 @@ def destroy
     @question = Question.find_by!(uid: params[:question_id])
   end
 
+  private def params_as_hash
+    valid_params.is_a?(Array) ? valid_params.map {|x| x.to_h } : valid_params.to_h
+  end
+
   private def valid_params
-    params.require(:focus_point).except(:uid)
+    filtered_params = params.require(:focus_point)
+    filtered_params.is_a?(Array) ? filtered_params.map {|x| x.except(:uid).permit! } : filtered_params.except(:uid).permit!
```

Comment:
For consistency across methods, I would remove the method: `params_as_hash` and do the hash conversion in `valid_params`. 

```
private def valid_params
    filtered_params = params.require(:focus_point)

    if filtered_params.is_a?(Array)
      filtered_params.map { |x| x.except(:uid).permit!.to_h }
    else
      filtered_params.except(:uid).permit!.to_h
    end    
end
```